packages/extension-api/src/extension-api.d.ts

```diff
@@ -2481,16 +2481,27 @@ declare module '@podman-desktop/api' {
     displayName: string;
     markdownDescription: string;
     images: ProviderImages;
+    path: string;
 
     /**
-     * Within your extension, it is reccommended to implement your own functionality to check the current
+     * Within your extension, it is recommended to implement your own functionality to check the current
      * version number of the CLI tool. For example, parsing the information from the CLI tool's `--version` flag.
      * Passing in path will also help to show where the CLI tool is expected to be installed.
-     * This is usually the ~/.local/share/containers/podman-desktop/extensions-storage directory.
+     *
+     * You can either pass in the current version in the `version` field or implement `getLocalVersion` function to
+     * return the current version of the CLI tool.
+     *
      * Note: The expected value should not include 'v'.
      */
-    version: string;
-    path: string;
+
+    // Version is optional if getLocalVersion is implemented
+    version?: string;
+
+    /**
+     * Since it's difficult to determine how to retrieve the exact version of the local binary, we recommend the
+     * extension use the getLocalVersion function to retrieve the version of the CLI tool.
+     */
+    getLocalVersion: () => Promise<string>;
```

Comment:
The comment above `version` says `// Version is optional if getLocalVersion is implemented` but getLocalVersion is always implemented. Should it be `getLocalVersion?:`?

Reply:
Yes as pointed out by @benoitf if we want to "remove" something, we make it optional and going forward just have `getLocalVersion` since every CLI provider should implement a way to retrieve the local binary version. I'll fix the comment.